src/sentry/api/endpoints/organization_events_trace.py

```diff
@@ -557,24 +568,42 @@ def augment_transactions_with_spans(
     # Fetch parent span ids of segment spans and their corresponding
     # transaction id so we can link parent/child transactions in
     # a trace.
-    spans_params = params.copy()
-    spans_params["project_objects"] = [p for p in params["project_objects"] if p.id in projects]
-    spans_params["project_id"] = list(projects.union(set(problem_project_map.keys())))
-
-    parents_results = SpansIndexedQueryBuilder(
-        Dataset.SpansIndexed,
-        spans_params,
-        query=f"trace:{trace_id} span_id:[{','.join(query_spans)}]",
-        selected_columns=[
-            "transaction.id",
-            "span_id",
-            "timestamp",
-        ],
-        orderby=["timestamp", "id"],
-        limit=10000,
-    ).run_query(referrer=Referrer.API_TRACE_VIEW_GET_PARENTS.value)
 
-    parent_map = {parent["span_id"]: parent for parent in parents_results["data"]}
+    def load_results(chunked_project_ids):
+        spans_params = params.copy()
+        spans_params["project_objects"] = [
+            p for p in params["project_objects"] if p.id in chunked_project_ids
+        ]
+        spans_params["project_id"] = chunked_project_ids
+        return SpansIndexedQueryBuilder(
+            Dataset.SpansIndexed,
+            spans_params,
+            query=f"trace:{trace_id} span_id:[{','.join(query_spans)}]",
+            selected_columns=[
+                "transaction.id",
+                "span_id",
+                "timestamp",
+            ],
+            orderby=["timestamp", "id"],
+            limit=10000,
+        ).run_query(referrer=Referrer.API_TRACE_VIEW_GET_PARENTS.value)
+
+    parent_map = {}
+    chunk_size = math.ceil(len(projects) / MAX_PARALLEL_QUERIES)
+    # Limit to 10 projects per query
```

Comment:
Won't this result in more than 3 queries if there are >30 projects

Reply:
yeah, just ensuring we do 3 in parallel, and anything more happens serially

Reply:
Hmm this worries me for orgs with many projects, could we just split everything in 3 for now and see what the performance boost is like?